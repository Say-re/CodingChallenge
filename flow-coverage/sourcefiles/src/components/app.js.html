<!DOCTYPE html>
<html><head><meta charSet="utf-8"/><link rel="stylesheet" href="../../../assets/semantic.min.css"/><link rel="stylesheet" href="../../../assets/codemirror.css"/><link rel="stylesheet" href="../../../assets/flow-highlight-source.css"/><link rel="stylesheet" href="../../../assets/flow-coverage-report.css"/><link rel="stylesheet" href="../../../assets/codemirror-simplescrollbars-addon.css"/><script src="../../../assets/jquery-3.1.0.min.js"></script><script src="../../../assets/semantic.min.js"></script><script src="../../../assets/semantic-tablesort.js"></script><script src="../../../assets/codemirror.js"></script><script src="../../../assets/codemirror-javascript-mode.js"></script><script src="../../../assets/codemirror-annotatescrollbar-addon.js"></script><script src="../../../assets/codemirror-simplescrollbars-addon.js"></script><script src="../../../assets/flow-highlight-source.js"></script><script src="../../../assets/index.js"></script><style>
.ui .CodeMirror {
  border: 1px solid rgba(34,36,38,.15);
  height: auto;
}
</style></head><body><div class="ui grid container"><div class="row"><div class="twelve wide column"><h2 class="ui header"><a href="../../../index.html" id="link-to-summary">Flow Coverage Report</a></h2></div></div><div class="row"><table class="ui small celled unstackable table"><thead><tr><th class="sorted ascending">Filename</th><th>Annotation</th><th>Percent</th><th>Total</th><th>Covered</th><th>Uncovered</th></tr></thead><tbody><tr class="negative"><td class="">src/components/app.js</td><td> @flow </td><td class=""><span>54 %</span></td><td> 11 </td><td> 6 </td><td> 5 </td></tr></tbody></table></div><div class="row ui one column centered grid"><div class="column" style="text-align:left"><div class="row"><div class="ui form"><div class="fields"><div class="sixteen wide inline field"><select class="ui search dropdown uncovered-locations"><option value="">Uncovered Locations</option><option value="4">Start: 4,8 - End: 4,14</option><option value="5">Start: 5,8 - End: 5,26</option><option value="6">Start: 6,10 - End: 6,22</option><option value="6">Start: 6,25 - End: 6,30</option><option value="6">Start: 6,33 - End: 6,37</option></select></div><div class="four wide inline field"><select class="ui search dropdown syntax-highlighting"><option value="es">ES6/ES7</option><option value="js">JavaScript</option><option value="no">None</option></select></div></div></div></div><textarea readonly="" id="file-content">// @flow

import React from &#x27;react&#x27;;
import JobForm from &#x27;../containers/form&#x27;;
import ApplicationResponse from &#x27;../containers/applicationResponse&#x27;;
import { BrowserRouter, Routes, Route } from &#x27;react-router-dom&#x27;;

export const App = () =&gt; {
  return (
    &lt;BrowserRouter&gt;
      &lt;Routes&gt;
        &lt;Route path=&quot;/&quot; element={&lt;JobForm /&gt;} /&gt;
        &lt;Route path=&quot;form&quot; element={&lt;ApplicationResponse /&gt;} /&gt;
      &lt;/Routes&gt;
    &lt;/BrowserRouter&gt;

  )
};</textarea></div></div><div class="row centered"><footer>Flow Coverage Report generated by <a href="https://flowtype.org">flow</a> and <a href="https://github.com/rpl/flow-coverage-report">flow-coverage-report</a> at Wed Dec 22 2021 23:29:23 GMT-0800 (Pacific Standard Time)</footer></div></div><pre id="file-coverage-data" style="display:none">{&quot;expressions&quot;:{&quot;covered_count&quot;:6,&quot;covered_locs&quot;:[{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:3,&quot;column&quot;:8,&quot;offset&quot;:17},&quot;end&quot;:{&quot;line&quot;:3,&quot;column&quot;:12,&quot;offset&quot;:22}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:8,&quot;column&quot;:14,&quot;offset&quot;:227},&quot;end&quot;:{&quot;line&quot;:8,&quot;column&quot;:16,&quot;offset&quot;:230}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:8,&quot;column&quot;:20,&quot;offset&quot;:233},&quot;end&quot;:{&quot;line&quot;:18,&quot;column&quot;:1,&quot;offset&quot;:443}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:10,&quot;column&quot;:5,&quot;offset&quot;:256},&quot;end&quot;:{&quot;line&quot;:15,&quot;column&quot;:20,&quot;offset&quot;:436}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:12,&quot;column&quot;:34,&quot;offset&quot;:320},&quot;end&quot;:{&quot;line&quot;:12,&quot;column&quot;:44,&quot;offset&quot;:331}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:13,&quot;column&quot;:37,&quot;offset&quot;:372},&quot;end&quot;:{&quot;line&quot;:13,&quot;column&quot;:59,&quot;offset&quot;:395}}],&quot;uncovered_count&quot;:5,&quot;uncovered_locs&quot;:[{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:4,&quot;column&quot;:8,&quot;offset&quot;:44},&quot;end&quot;:{&quot;line&quot;:4,&quot;column&quot;:14,&quot;offset&quot;:51}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:5,&quot;column&quot;:8,&quot;offset&quot;:86},&quot;end&quot;:{&quot;line&quot;:5,&quot;column&quot;:26,&quot;offset&quot;:105}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:6,&quot;column&quot;:10,&quot;offset&quot;:157},&quot;end&quot;:{&quot;line&quot;:6,&quot;column&quot;:22,&quot;offset&quot;:170}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:6,&quot;column&quot;:25,&quot;offset&quot;:172},&quot;end&quot;:{&quot;line&quot;:6,&quot;column&quot;:30,&quot;offset&quot;:178}},{&quot;source&quot;:&quot;/Users/Sayre/coding/CodingChallenge/src/components/app.js&quot;,&quot;type&quot;:&quot;SourceFile&quot;,&quot;start&quot;:{&quot;line&quot;:6,&quot;column&quot;:33,&quot;offset&quot;:180},&quot;end&quot;:{&quot;line&quot;:6,&quot;column&quot;:37,&quot;offset&quot;:185}}],&quot;empty_count&quot;:0,&quot;empty_locs&quot;:[]},&quot;filename&quot;:&quot;src/components/app.js&quot;,&quot;annotation&quot;:&quot;flow&quot;,&quot;isFlow&quot;:true,&quot;percent&quot;:54}</pre></body></html>