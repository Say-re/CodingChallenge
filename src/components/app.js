// @flow

import React from 'react';
import JobForm from '../containers/form';
import ApplicationResponse from '../containers/applicationResponse';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

export const App = () => {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<JobForm />} />
        <Route path="form" element={<ApplicationResponse />} />
      </Routes>
    </BrowserRouter>

  )
};