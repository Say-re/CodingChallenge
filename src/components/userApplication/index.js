// @flow

import React, { Fragment } from 'react';
import { FieldWrapper, PartOneWrapper, FieldSubWrapper, FieldSubWrapper2, BlockFormFieldsWrapper } from '../../styled-components/forms/form-wrappers'
import type { InputProps } from '../../lib/types/forms';

const UserApplication = ({ firstName, lastName, userEmail, userAddress, userCity, userState, userZipcode, userTelephone, userJob, userExperience, userTechnologyExperience, userShortAnswer, userTest }: InputProps) => {
  return (
    <Fragment>
      <PartOneWrapper>
        <FieldWrapper>
          <FieldSubWrapper>
            <p className="resLabel">First Name</p>
            <p className="resField">{firstName}</p>
          </FieldSubWrapper>
          <FieldSubWrapper>
            <p className="resLabel">Last Name</p>
            <p className="resField">{lastName}</p>
          </FieldSubWrapper>
          <FieldSubWrapper>
            <p className="resLabel">Email</p>
            <p className="resField">{userEmail}</p>
          </FieldSubWrapper>
        </FieldWrapper>
        <FieldWrapper>
          <FieldSubWrapper2>
            <p className="resLabel">Address</p>
            <p className="resField">{userAddress}</p>
          </FieldSubWrapper2>
          <FieldSubWrapper>
            <p className="resLabel">Phone Number</p>
            <p className="resField">{userTelephone}</p>
          </FieldSubWrapper>
        </FieldWrapper>
        <FieldWrapper>
          <FieldSubWrapper>
            <p className="resLabel">City</p>
            <p className="resField">{userCity}</p>
          </FieldSubWrapper>
          <FieldSubWrapper>
            <p className="resLabel">State</p>
            <p className="resField">{userState}</p>
          </FieldSubWrapper>
          <FieldSubWrapper>
            <p className="resLabel">Zipcode</p>
            <p className="resField">{userZipcode}</p>
          </FieldSubWrapper>
        </FieldWrapper>
      </PartOneWrapper>
      <BlockFormFieldsWrapper>
        <p className="resLabel">Job Title</p>
        <p className="resField">{userJob}</p>
        <p className="resLabel">Number of Years Developing</p>
        <p className="resField">{userExperience}</p>
        <p className="resLabel">Technology Experience</p>
        <div className="resField">
          {userTechnologyExperience.map((technology: string, index: number) => {
            return <p key={index}>{technology}</p>
          })}
        </div>
        <p className="resLabel">Describe why you are a good fit for the job you are applying for</p>
        <p className="resField">{userShortAnswer}</p>
        <p className="resLabel">I acknowledge that this is only a test</p>
        <p className="resField">{userTest}</p>
      </BlockFormFieldsWrapper>
    </Fragment>
  )
}

export default UserApplication;