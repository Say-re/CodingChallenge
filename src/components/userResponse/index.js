import React from 'react';
import { connect } from 'react-redux';

const UserResponse = ({ firstName }) => {
  return (
    <div>
      <h1>Intelage Application Successful</h1>
      <h2>This is going to work!!</h2>
    </div>
  )
};

const mapStateToProps = (state) => ({
  ...state,
})

export default connect(mapStateToProps, null)(UserResponse)