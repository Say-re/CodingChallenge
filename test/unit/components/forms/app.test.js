import React from 'react';
import { shallow } from 'enzyme';
import { App } from '../../../../src/components/app'

describe('App Component Unit Tests', () => {
  test('AppComponent Snapshot Test', async () => {
    const wrapper = shallow(<App />)
    await expect(wrapper).toMatchSnapshot();
  })
})