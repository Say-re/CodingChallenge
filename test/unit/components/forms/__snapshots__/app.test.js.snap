// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Unit Tests AppComponent Snapshot Test 1`] = `
<BrowserRouter>
  <Routes>
    <Route
      element={<Memo(Connect(JobForm)) />}
      path="/"
    />
    <Route
      element={<Memo(Connect(ApplicationResponse)) />}
      path="form"
    />
  </Routes>
</BrowserRouter>
`;
